<launch>
  <arg name="gui" default="true"/>

  <node pkg="jsk_interactive_marker" type="marker_6dof" name="sensor_frame">
    <rosparam>
      publish_tf: true
      tf_frame: sensor_frame
      frame_id: world
      publish_pose_periodically: true
      object_x: 0.1
      object_y: 0.1
      object_z: 0.1
      initial_z: 0.0
    </rosparam>
  </node>
  <node pkg="jsk_interactive_marker" type="marker_6dof" name="target_frame">
    <rosparam>
      publish_tf: false
      frame_id: world
      publish_pose_periodically: true
      object_x: 0.1
      object_y: 0.1
      object_z: 0.1
      initial_x: 0.24
      initial_z: 1.8
    </rosparam>
  </node>
  <node pkg="jsk_interactive_marker" type="dummy_camera.py" name="dummy_camera">
    <rosparam>
      frame_id: sensor_frame
    </rosparam>
  </node>
  <node pkg="jsk_pcl_ros" type="sample_boundingbox_occlusion_rejector.py" name="sample_boundingbox_occlusion_rejector">
    <remap from="~input" to="dummy_camera/camera_info" />
    <remap from="~input/candidate_pose" to="target_frame/pose" />
  </node>
  <node pkg="jsk_pcl_ros" type="tf_transform_bounding_box_array" name="sensor_target">
    <remap from="~input" to="sample_boundingbox_occlusion_rejector/output/target_boxes" />
    <rosparam>
      target_frame_id: sensor_frame
    </rosparam>
  </node>
  <node pkg="jsk_pcl_ros" type="tf_transform_bounding_box_array" name="sensor_candidate">
    <remap from="~input" to="sample_boundingbox_occlusion_rejector/output/candidate_boxes" />
    <rosparam>
      target_frame_id: sensor_frame
    </rosparam>
  </node>
  <node pkg="jsk_pcl_ros" type="boundingbox_occlusion_rejector" name="boundingbox_occlusion_rejector">
    <remap from="~input/camera_info" to="dummy_camera/camera_info" />
    <remap from="~input/target_boxes" to="sensor_target/output" />
    <remap from="~input/candidate_boxes" to="sensor_candidate/output" />
  </node>

  <group if="$(arg gui)">
    <node pkg="rviz" type="rviz" name="rviz"
          args="-d $(find jsk_pcl_ros)/config/sample_boundingbox_occlusion_rejector.rviz" />
  </group>
</launch>
